### PR TITLE
ESC-351 send nil return on creation to set last usage date to creation

### PR DESCRIPTION
**Pull request by sam-miele**

### CODE REVIEW

**Review by carwynellis — APPROVED** [review]
LGTM 👍

File: app/uk/gov/hmrc/eusubsidycompliance/controllers/UndertakingController.scala [hunk 1]
```diff
@@ -49,9 +52,10 @@ class UndertakingController @Inject()(
   def create: Action[JsValue] = Action.async(parse.json) { implicit request =>
     implicit val uF = Json.format[Undertaking]
     withJsonBody[Undertaking] { undertaking: Undertaking =>
-      eis.createUndertaking(undertaking).map{ ref =>
-        Ok(Json.toJson(ref))
-      }
+      for {
+        ref <- eis.createUndertaking(undertaking)
```
Comment (priyankadhiman3006) [inline app/uk/gov/hmrc/eusubsidycompliance/controllers/UndertakingController.scala:56]: one quick question may be the dumb one, when an undertaking is created , then a subsidy is also created? or updateSubsidy do insert and update?

Comment (sam-miele) [reply app/uk/gov/hmrc/eusubsidycompliance/controllers/UndertakingController.scala:56]: UpdateSubsidy will insert it if it's not there. We could rename to upsert?

Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/eusubsidycompliance/controllers/UndertakingController.scala:56]: sure, if you like I just wanted to confirm that :)

**Review by priyankadhiman3006 — APPROVED** [review]

**Pull request merged**